packages/cli/lib/s3.ts

```diff
@@ -29,3 +30,33 @@ export async function openFile (this: any, csvw: Csvw.Mapping, s3Endpoint: strin
 
   throw new Error(`Could not read file "${csvw.url}" from S3. It was neither Buffer or Readable`)
 }
+
+export function uploadFile (this: any, fileName: string, s3Endpoint: string, s3Bucket: string) {
```

Comment:
Could the first argument be typed as `Context` instead of `any`?

By the way, `this` is a confusing name. I'm guessing it's a convention that's used for all pipeline functions?

Reply:
No, this is typescript's annotation of the function... call context.

This is called by `barnard59` which does not have type declarations. Would have to add `.d.ts` for it and use the right type here.

Reply:
https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters

Reply:
Ok so it's not the same `this` as the one in https://github.com/zazuko/data-cube-curation/blob/master/packages/cli/lib/project.ts#L88 ?

Reply:
Thanks, I didn't know about Typescript's `this`

Reply:
So I forgot that I'd already added a declaration for the `Context`? 😊 

Of course, will use that